### PR TITLE
Fix `Mutex_m#initialize` when the super's initialize has kwargs

### PR DESCRIPTION
**Pull request by pocke**
This pull request fixes an error of Ruby 3 keyword arguments for `Mutex_m#initialize`.

See https://github.com/ruby/ruby/pull/3310

### CODE REVIEW

**Comment by pocke** [comment]
I've updated the patch because mutex_m library supports old Rubies but the previous patch uses `...` syntax.

File: lib/mutex_m.rb [hunk 1]
```diff
@@ -109,8 +109,17 @@ def mu_initialize # :nodoc:
...
+      super
+    end
+  else
+    eval <<~RUBY, binding, __FILE__, __LINE__ + 1
+      def initialize(...) # :nodoc:
+        mu_initialize
+        super
+      end
+    RUBY
   end
```
Comment (kamipo) [inline lib/mutex_m.rb:124]: As an alternative, we can use `ruby2_keywords` instead.

```suggestion
  end
  ruby2_keywords(:initialize) if respond_to?(:ruby2_keywords, true)
```

Comment (jeremyevans) [reply lib/mutex_m.rb:124]: I agree with @kamipo that `ruby2_keywords` is the best way to handle this.

Comment (pocke) [reply lib/mutex_m.rb:115]: Thanks for your suggestion! I applied the patch in https://github.com/ruby/mutex_m/pull/4/commits/f55360b7399086e3adaa56e2d409e4e5d23c2e77

**Comment by pocke** [comment]
Any news? I guess Ruby 3 needs this patch because Mutex_m will not work well without this patch in Ruby 3.
if we need to do something for this patch, please tell me it. I'll work on it :muscle:

**Comment by jeremyevans** [comment]
@pocke I don't think this needs anything else.  I'll merge this now.

**Pull request merged**